library/std/src/sys/uefi/alloc.rs

```diff
@@ -0,0 +1,28 @@
+//! Global Allocator for UEFI.
+//! Uses [r-efi-alloc](https://crates.io/crates/r-efi-alloc)
+
+use crate::alloc::{handle_alloc_error, GlobalAlloc, Layout, System};
+
+const MEMORY_TYPE: u32 = r_efi::efi::LOADER_DATA;
+
+#[stable(feature = "alloc_system_type", since = "1.28.0")]
+unsafe impl GlobalAlloc for System {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let system_table = match crate::os::uefi::env::try_system_table() {
+            None => return crate::ptr::null_mut(),
+            Some(x) => x.as_ptr() as *mut _,
+        };
+
+        // The caller must ensure non-0 layout
+        unsafe { r_efi_alloc::raw::alloc(system_table, layout, MEMORY_TYPE) }
```

Comment:
UEFI is one of the targets where there will be a lot of code that will not use std even if it technically could. 

[r-efi-alloc](https://github.com/r-efi/r-efi-alloc) serves as a common allocator for UEFI. The `alloc` here is not just a call to `alloc` function. Internally, it has to do some windows allocator stuff to support the allocation of greater than 8 alignments.

Initially, this code was present in std, but after asking in the full PR, I decided to extract it into this separate crate. I can move this stuff back here if it is needed.

Reply:
Hmm. That doesn't seem like too much code to put into std but I don't think it's absolutely necessary to have it in std either... it can always be upstreamed in a separate PR. So "big enough", yes. That's fair.